pkg/controllers/vdb/onlineupgrade_reconciler.go

```diff
@@ -682,49 +693,47 @@ func (r *OnlineUpgradeReconciler) postPauseConnectionsMsg(ctx context.Context) (
-	// repurpose this step to close all existing sessions. We forcibly close all
-	// connections as we want to prevent writes from happening. Continuing to
-	// allow writes could potentially lead to data loss. We are about to
-	// replicate the data, if writes can happen after the replication to replica
-	// group B, they are going to be lost.
-	//
-	// We first need to route all traffic away from all subclusters in replica
-	// group A. There is no target they will get routed too. Clients just won't
-	// be able to connect until we finish the fail-over to replica group B. We
-	// want do this for all pods that are in replica group A. The pod facts that
-	// we pass in are for the main cluster, so that covers the pods we want to
-	// do this for.
-	actor := MakeClientRoutingLabelReconciler(r.VRec, r.Log, r.VDB, r.PFacts[vapi.MainCluster], DrainNodeApplyMethod, "")
-	r.Manager.traceActorReconcile(actor)
-	res, err := actor.Reconcile(ctx, &ctrl.Request{})
-	if verrors.IsReconcileAborted(res, err) {
-		return res, err
+	pf := r.PFacts[vapi.MainCluster]
+	initiator, ok := pf.findFirstUpPod(false /*not allow read-only*/, "" /*arbitrary subcluster*/)
+	if !ok {
+		r.Log.Info("No Up nodes found. Requeue reconciliation.")
+		return ctrl.Result{Requeue: true}, nil
 	}
 
-	// close all existing user sessions
-	err = r.Manager.closeAllSessions(ctx, r.PFacts[vapi.MainCluster])
-	if err != nil {
-		return ctrl.Result{}, err
+	err := r.Dispatcher.ManageConnectionDraining(ctx,
+		manageconnectiondraining.WithInitiator(initiator.podIP),
+		manageconnectiondraining.WithAction(vclusterops.ActionPause),
+	)
+
+	return ctrl.Result{}, err
+}
+
+func (r *OnlineUpgradeReconciler) waitForConnectionsPaused(ctx context.Context) (ctrl.Result, error) {
+	if vmeta.GetOnlineUpgradeStepInx(r.VDB.Annotations) > waitForConnectionsPauseInx {
+		return ctrl.Result{}, nil
 	}
 
-	// Iterate through the subclusters in replica group A. We check if there are
-	// any active connections for each. Once they are all idle we can advance to
-	// the next action in the upgrade.
-	scNames := r.VDB.GetSubclustersForReplicaGroup(vmeta.ReplicaGroupAValue)
-	for _, scName := range scNames {
-		res, err := r.Manager.isSubclusterIdle(ctx, r.PFacts[vapi.MainCluster], scName)
-		if verrors.IsReconcileAborted(res, err) {
-			return res, err
-		}
+	pfacts := r.PFacts[vapi.MainCluster]
+	_, ok := pfacts.findFirstUpPod(false /*not allow read-only*/, "" /*arbitrary subcluster*/)
+	if !ok {
+		r.Log.Info("No Up nodes found; Requeue reconciliation")
+		return ctrl.Result{Requeue: true}, nil
 	}
 
-	return ctrl.Result{}, nil
+	if res, err := r.Manager.areAllConnectionsPaused(ctx, pfacts); err != nil {
+		r.Log.Error(err, "Error checking if connections are paused; Requeue reconciliation")
+		return ctrl.Result{Requeue: true}, nil
+	} else if res.Requeue {
+		r.Log.Error(err, "Cluster still has active connections. Requeue reconciliation")
+		return ctrl.Result{Requeue: true}, nil
+	}
```

Comment:
```suggestion
res, err := r.Manager.areAllConnectionsPaused(ctx, pfacts)
if verrors.IsReconcileAborted(res, err) {
	return res, err
}
```

You do not need to log those errors there.